lib/engine/game/g_18_bf/game.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+require_relative '../g_1867/game'
+require_relative 'meta'
+require_relative 'entities'
+require_relative 'map'
+require_relative 'market'
+require_relative 'tiles'
+require_relative 'trains'
+
+module Engine
+  module Game
+    module G18BF
+      class Game < G1867::Game
+        include Entities
+        include Map
+        include Market
+        include Tiles
+        include Trains
+
+        include_meta(G18BF::Meta)
+
+        GAME_END_REASONS_TEXT = Base::GAME_END_REASONS_TEXT.merge(
+          custom: 'The first 4+4 or 6G train is purchased.',
+        )
+        GAME_END_CHECK = { custom: :one_more_full_or_set }.freeze
+
+        def setup
+          # TODO: check which bits of this are needed, just cut-n-pasted from 1867.
+          @interest = {}
+          setup_company_price_up_to_face
+
+          @show_majors = false
+          setup_london_hexes
+          setup_bonuses
+        end
+
+        def setup_preround
+          # Randomise the privates available and the minor company order.
+          random = Random.new(rand)
```

Comment:
why are you using Random? you can't do that, you can only use the rand method in the game as that works on both server and client

Reply:
I'd misunderstood the use of the Game.rand method. I've now removed the use of the Ruby Random class, and I'm instead using `Array.sort_by { rand }`.